source/Ship.cpp

```diff
@@ -2605,6 +2605,10 @@ void Ship::Disable()
 void Ship::Destroy()
 {
 	hull = -1.;
+	
+	shared_ptr<Ship> oldParent = parent.lock();
+	if(oldParent)
+		oldParent->TuneForEscorts();
```

Comment:
If you want: In `Ship::Move` the ship checks whether it's dead and then does various death related stuff. If you put there `SetParent(nullptr);`, you'll get this behavior for free, since `SetParent` removes escorts from the old parent.

Reply:
Ships can also be destroyed due to a Mission-Action. So moving it there might cause that case to be missed. (But I guess I could replace this code here by `SetParent(nullptr)`.

Reply:
Person ships are recylced (using the Ship->Restore() function), so it looks like we are not allowed to break the parent/child relations on destroyed ships (at least not on person ships).

Reply:
Ah didn't know that, thanks!